OpenRA.Mods.Cnc/Traits/GpsWatcher.cs

```diff
@@ -110,5 +110,15 @@ public void UnregisterForOnGpsRefreshed(Actor actor, IOnGpsRefreshed toBeNotifie
 		{
 			notifyOnRefresh.Remove(new TraitPair<IOnGpsRefreshed>(actor, toBeNotified));
 		}
+
+		protected override void TraitEnabled(Actor self)
+		{
+			self.World.AddFrameEndTask(w => GpsAdd(self));
```

Comment:
This looks a bit suspicious to me. Just to clarify, `self` is the player actor here, right? However, `GpsAdd` should be called with the actor that has the `GpsPower` trait if I read this correct. I suppose the current code would still work even if we have a bogus parameter, but we should really sort that out if my suspicions are correct.

Reply:
This just needs to trigger the `Granted` check, so I changed it accordingly.